### ISSUE TITLE
Sort Colors into the correct names / Figure out a way to fix contrast

### ISSUE THREAD

**Issue by s-ol** [s-ol/themer#25]
Either sort colors to the nearest correct name (`red`, `green`, `bright_blue` etc.) or find a way to label the colors so that there are obvios ways to use them and have readable contrast; for example sorting them in pairs (`color1_dark`, `color1_light`).

**Comment by Kaligule** [comment]
[Interesting article about the subject](http://www.alanzucconi.com/2015/09/30/colour-sorting/)

I guess it might be enought to compare the RGBvalues.

**Comment by ghost** [comment]
can i work on this? is there a deadline?
i'm new to contributing

**Comment by Kaligule** [comment]
Since the issue is 1.5 years old and has the  label _up-for-grabs_ I think you could just go for it. Do you know how you want to handle this?

**Comment by ghost** [comment]
not really, i would like to understand more about the bug, i asked because of the k means parser label

**Comment by s-ol** [comment]
@shreab373 you can definetely contribute, but this is not a bug but a rather major feature. It is `up-for-grabs` because I don't know a good way of doing it, but there are other issues (the recent ones for the 2.0 milestone) that are much easier to implement if you want to get started on something.

If you used themer you probably noticed that lots of colorschemes selected from images are not very useful with default terminal configs, because the contrast doesn't really work out. Basically what is needed would be to either sort the colors so that there is a way to use any themer colorscheme and always have some contrast (and ideally even match the color names more or less, at least dark/bright).

Another problem is that sometimes the colors themselves just have no contrast, so maybe the color generation needs to be adjusted too.

**Comment by ghost** [comment]
but not sure if i can do this, bu thank you letting me know that its a feature:)

**Comment by zybex86** [comment]
Is this issue still being worked on? I'm more of a WebDev but learning Python (learned JS, Java, C#). I would like to try and work on this :)

**Comment by s-ol** [comment]
@zybex86 I am not aware of anyone pursuing this currently, if you want to investigate this I'd be happy to see what you come up with :)

**Comment by zybex86** [comment]
I'm still looking into this, but I don't think it's a beginner-friendly issue ;)

**Comment by zybex86** [comment]
I am still having problems using themer to test how it works. For the time being I will drop this issue - will come back to it when I get more experienced with themer
